### ISSUE TITLE
TensorRT model inference speed varies dramatically depending on if an ONNX model is also loaded

### ISSUE THREAD

**Issue by wreck-a-nice-beach** [triton-inference-server/server#5938]
**Description**
I observe very different throughput and latency for my TensorRT model depending on whether or not an ONNX model is also loaded on the server. When I load only the TensorRT model, or when all models that are loaded are TensorRT, then I observe much lower throughput and higher latency than when an ONNX model is also loaded on the server. The ONNX model need not be anything related. I see the same behavior if the ONNX model is just the densenet_onnx model from the Triton Server examples. I am measuring throughput and latency using the performance analyzer tool. Throughput of the TensorRT model is about 3x higher when an ONNX model is also loaded.  I don't understand why it is so much faster when an ONNX model is loaded unless the onnxruntime backend has something to do with it.

**Triton Information**
I am using  nvcr.io/nvidia/tritonserver:23.05-py

**To Reproduce**
I start from a Pytorch model and export to ONNX. I build the tensorrt engine and serialize it starting from the ONNX model in python. The config.pbtxt for the model looks like this:

name: "encoder_trt"
platform: "tensorrt_plan"
max_batch_size: 16
input {
  name: "feature_window"
  data_type: TYPE_FP32
  dims: [80, -1]
}
input {
  name: "feature_length"
  data_type: TYPE_INT32
  dims: [1]
  reshape: { shape: [ ] }
}
output {
  name: "logits"
  data_type: TYPE_FP32
  dims: [-1, 129]
}
instance_group {
  count: 1
  kind: KIND_GPU
}
dynamic_batching {
  preferred_batch_size: [8, 16]
  max_queue_delay_microseconds: 300
  preserve_ordering: true
}

When only my TensorRT model is loaded, Triton reports the tensorrt backend loaded:
`tensorrt | /opt/tritonserver/backends/tensorrt/libtriton_tensorrt.so | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability":"6.000000","default-max-batch-size":"4"}}`

And when both ONNX models and TensorRT models are loaded:
`tensorrt    | /opt/tritonserver/backends/tensorrt/libtriton_tensorrt.so       | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability":"6.000000","default-max-batch-size":"4"}} |
| onnxruntime | /opt/tritonserver/backends/onnxruntime/libtriton_onnxruntime.so | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability":"6.000000","default-max-batch-size":"4"}}`

This part seems as expected.

**Expected behavior**
I expect the performance of the TensorRT model not to be worse when an ONNX model is not also loaded on the server

**Comment by wreck-a-nice-beach** [comment]
closing this because I no longer see any discrepancy once I use trtexec command in the tensorrt 23.05 container to create the model.plan file

**Comment by wreck-a-nice-beach** [comment]
closing this because I no longer see any discrepancy once I use trtexec command in the tensorrt 23.05 container to create the model.plan file
